lib/spack/spack/test/cmd/compiler.py

```diff
@@ -103,3 +188,47 @@ def test_compiler_add(
     new_compiler = new_compilers - old_compilers
     assert any(c.version == spack.version.Version(mock_compiler_version)
                for c in new_compiler)
+
+
+def test_compiler_add_multiple_in_path(
+    mock_two_compiler_dirs, mutable_empty_config, mock_compiler_version
+):
+    # Find compilers
```

Comment:
This `#` comment is unnecessary given that `compiler_find` is called right after. The test could use a docstring though (something like "When two compilers are detected with the same spec, make sure Spack uses the first one that appears in the ordered list of paths provided").